docs/spec.md

```diff
@@ -6485,6 +6533,12 @@ function returns `true`. If `x` is not a tensor, returns `None`.
 tensors and returns `num_results` slices of `x` along the axis `axis`.
 If `x` is not a tensor or `dim(x, axis) % num_results != 0`, returns `None`.
 
+#### Functions on non-values
+
+* `is_defined_in_parent_scope(x: Function) -> Value` is defined on functions
```

Comment:
`string -> Value`? I agree that we should resolve the `Value` issue separately, and resolve the `string` vs `FunctionRef` issue separately.

Reply:
Also for history - This lack of string vs function distinction has bit us before when using discardable attributes. We only support `@func` in custom_call and Func, anywhere else will fail to serialize, and it isn't trivial to add because it is ambiguous with any string.

Reply:
Not too late, and probably not a bad idea to do before v1.0. Just need to manage forward/backward compat. Shouldn't be too difficult